pkg/util/metadata/metadata.go

```diff
@@ -77,6 +77,15 @@ type Metadata struct {
 	// .. and other fields we don't care about.  Expand as necessary.
 }
 
+type IMetadata interface {
+	GetInstanceID(order string) (string, error)
+	GetAvailabilityZone(order string) (string, error)
+	Get(order string) (*Metadata, error)
+	GetFromMetadataService(metadataVersion string) (*Metadata, error)
+	GetFromConfigDrive(metadataVersion string) (*Metadata, error)
+	GetDefaultMetadataSearchOrder() string
```

Comment:
IMO GetDefaultMetadataSearchOrder(), it can be a utility method only need not to be part of interface

Reply:
I need double check this and see whether it's ok to do so, some place seems rely on this function